mdata/cache/ccache_metric.go

```diff
@@ -137,24 +266,29 @@ func (mc *CCacheMetric) generateKeys() {
 // assumes we already have at least a read lock
 func (mc *CCacheMetric) nextTs(ts uint32) uint32 {
 	chunk := mc.chunks[ts]
-	span := chunk.Itgen.Span
+	return mc.nextTsCore(chunk.Itgen, chunk.Ts, chunk.Prev, chunk.Next)
+}
+
+// nextTsCore returns the ts of the next chunk, given a chunks key properties
+// (to the extent we know them). It guesses if necessary.
+// assumes we already have at least a read lock
+func (mc *CCacheMetric) nextTsCore(itgen chunk.IterGen, ts, prev, next uint32) uint32 {
+	span := itgen.Span
 	if span > 0 {
 		// if the chunk is span-aware we don't need anything else
-		return chunk.Ts + span
+		return ts + span
 	}
 
-	if chunk.Next == 0 {
-		if chunk.Prev == 0 {
-			// if a chunk has no next and no previous chunk we have to assume it's length is 0
-			return chunk.Ts
-		} else {
-			// if chunk has no next chunk, but has a previous one, we assume the length of this one is same as the previous one
-			return chunk.Ts + (chunk.Ts - chunk.Prev)
-		}
-	} else {
-		// if chunk has a next chunk, then that's the ts we need
-		return chunk.Next
+	// if chunk has a next chunk, then that's the ts we need
+	if next != 0 {
```

Comment:
If the chunk has a `.Next` ts and is also span aware, but `itgen.Ts + itgen.Span != itgen.Next` (a corruption), shouldn't we rather prioritize the `next` ts?  But then again, if the chunk has a `.Next` attribute, do we even need to call this method? I think if `nextTs()` is called and `chunk.Next > 0` then there is no need to call `nextTsCore()`

Reply:
> If the chunk has a .Next ts and is also span aware, but itgen.Ts + itgen.Span != itgen.Next (a corruption), shouldn't we rather prioritize the next ts?

i don't know. i just kept the logic you wrote. it seems to be one of those should-never-happen-anyway-so-not-really-important cases.  in such edge case, .Next should always be set based on a real chunk that we have, whereas `itgen.Ts + itgen.Span` may not be a ts of a chunk that we have, so I think your argument makes sense, but I rather not touch that in this PR because this PR needs to go out and be deployed.

to be clear, nextTsCore() kept all the same logic that nextTs had, it just allows the logic to be called on the "core attributes" without needing to have a chunk already in place.

Reply:
ok, fair enough